SS14.Labeller/Program.cs

```diff
@@ -1,350 +1,95 @@
-
-    var matcher = new Matcher();
-    matcher.AddInclude("**/*.rsi/*.png");            // Sprites
-    matcher.AddInclude("Resources/Maps/**/*.yml");   // Map
-    matcher.AddInclude("Resources/Prototypes/Maps/**/*.yml");
-    matcher.AddInclude("**/*.xaml*");                // UI
-    matcher.AddInclude("**/*.swsl");                 // Shaders
-    matcher.AddInclude("**/*.ogg");                  // Audio
-
-    var sprites = new Matcher().AddInclude("**/*.rsi/*.png");
-    var maps = new Matcher().AddInclude("Resources/Maps/**/*.yml").AddInclude("Resources/Prototypes/Maps/**/*.yml");
-    var ui = new Matcher().AddInclude("**/*.xaml*");
-    var shaders = new Matcher().AddInclude("**/*.swsl");
-    var audio = new Matcher().AddInclude("**/*.ogg");
-    var cs = new Matcher().AddInclude("**/*.cs");
-
-    if (sprites.Match(changedFiles).HasMatches)
-        await AddLabel(client, owner, repoName, number, labelChangesSprites);
-
-    if (maps.Match(changedFiles).HasMatches)
-        await AddLabel(client, owner, repoName, number, labelChangesMap);
-
-    if (ui.Match(changedFiles).HasMatches)
-        await AddLabel(client, owner, repoName, number, labelChangesUi);
-
-    if (shaders.Match(changedFiles).HasMatches)
-        await AddLabel(client, owner, repoName, number, labelChangesShaders);
-
-    if (audio.Match(changedFiles).HasMatches)
-        await AddLabel(client, owner, repoName, number, labelChangesAudio);
-
-    if (!cs.Match(changedFiles).HasMatches)
-        await AddLabel(client, owner, repoName, number, labelChangesNoCSharp);
-}
-
-string ToHmacSha256(byte[] data, string secret)
-{
-    var key = Encoding.UTF8.GetBytes(secret);
-    using var hmac = new HMACSHA256(key);
-    var hashBytes = hmac.ComputeHash(data);
-    return Convert.ToHexString(hashBytes).ToLowerInvariant();
-}
-
-async Task<List<string>> GetChangedFiles(HttpClient client, string owner, string repo, int prNumber)
-{
-    // TODO: Ratelimit? Might explode on big PRs???
-
-    var files = new List<string>();
-    var page = 1;
-    while (true)
-    {
-        var res = await client.GetAsync($"{githubApiBase}/repos/{owner}/{repo}/pulls/{prNumber}/files?per_page=100&page={page}");
-        if (!res.IsSuccessStatusCode)
-            break; // TODO: Logging?
-
-        var content = await res.Content.ReadAsStringAsync();
-        var json = JsonDocument.Parse(content);
-        var batch = json.RootElement.EnumerateArray().Select(f => f.GetProperty("filename").GetString()!).ToList();
-        if (batch.Count == 0) break;
-
-        files.AddRange(batch);
-        if (batch.Count < 100) break;
-
-        page++;
+                if (string.IsNullOrEmpty(githubEvent))
+                    return Results.BadRequest("Missing GitHub event.");
+
+                var request = context.Request;
+
+                using var memStream = new MemoryStream();
+                await request.Body.CopyToAsync(memStream);
+                var bodyBytes = memStream.ToArray();
```

Comment:
If you are already copying into a memory stream, I think design wise it would be better to pass the stream into the methods and not calling an extra .ToArray() here.

Reply:
Ah, fair enough